tests/pipeline_operations_test.py

```diff
@@ -1,14 +1,83 @@
 import unittest
 import pyspark
 
+from absl.testing import parameterized
+import apache_beam as beam
+import apache_beam.testing.test_pipeline as test_pipeline
+from apache_beam.testing.util import assert_that
+from apache_beam.testing.util import equal_to
+
+from pipeline_dp import DataExtractors
 from pipeline_dp.pipeline_operations import SparkRDDOperations
 from pipeline_dp.pipeline_operations import LocalPipelineOperations
+from pipeline_dp.pipeline_operations import BeamOperations
 
 
 class PipelineOperationsTest(unittest.TestCase):
     pass
 
 
+class BeamOperationsTest(parameterized.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.ops = BeamOperations()
+        cls.data_extractors = DataExtractors(
+            partition_extractor=lambda x: x[1],
+            privacy_id_extractor=lambda x: x[0],
+            value_extractor=lambda x: x[2])
+
+    def test_filter_by_key_must_not_be_none(self):
+        col = [(1, 6, 1), (2, 7, 1), (3, 6, 1), (4, 7, 1), (5, 8, 1)]
+        public_partitions = None
+        with self.assertRaises(TypeError):
+            result = self.ops.filter_by_key(col, public_partitions,
+                                            self.data_extractors,
+                                            "Public partition filtering")
+
+    @parameterized.parameters(
+        {'in_memory': True},
+        {'in_memory': False},
+    )
+    def test_filter_by_key_remove(self, in_memory):
+        col = [(1, 7, 1), (2, 19, 1), (3, 9, 1), (4, 11, 1), (5, 10, 1)]
+        public_partitions = [7, 9]
+        expected_result = [(7, (1, 7, 1)), (9, (3, 9, 1))]
+        if in_memory:
+            result = self.ops.filter_by_key(col, public_partitions,
+                                            self.data_extractors,
+                                            "Public partition filtering")
+            self.assertEqual(result, expected_result)
+        else:
+            with test_pipeline.TestPipeline() as p:
+                pcol = (p | beam.Create(col))
```

Comment:
in_memory is only about public_partitions, please replace pcol -> public_partitions.

And then parametrized tests will be simplified, this line will be the only difference in tests.

Reply:
Got it. Thanks. Done.